notebooks/403-action-recognition-webcam/403-action-recognition-webcam.ipynb

```diff
@@ -0,0 +1,691 @@
+    "        encoder_output = []\n",
+    "        decoded_labels = [0, 0, 0]\n",
+    "        decoded_top_probs = [0, 0, 0]\n",
+    "        while True:\n",
+    "            \n",
+    "            encoder_output = []\n",
+    "            # loop over the number of required sample frames\n",
+    "            # Prepare a set of frames to be enconded-decoded\n",
+    "            for i in range(0, sample_duration):\n",
+    "                # read a frame from the video stream\n",
+    "                frame = player.next()\n",
+    "                frames.append(frame)\n",
+    "                \n",
+    "                if frame is None:\n",
+    "                    print(\"Source ended\")\n",
+    "                    break\n",
+    "                scale = 1280 / max(frame.shape)\n",
+    "                \n",
+    "                # Adaptative resize for visualization\n",
+    "                if scale < 1:\n",
+    "                    frame = cv2.resize(frame, None, fx=scale, fy=scale, interpolation=cv2.INTER_AREA)\n",
+    "                \n",
+    "                # Preprocess frame before Encoder\n",
+    "                preprocessed = preprocessing(frame, size)\n",
+    "                                \n",
+    "                # Measure processing time\n",
+    "                start_time = time.time()\n",
+    "                \n",
+    "                # Encoder Inference per frame\n",
+    "                encoder_output = encoder(preprocessed, exec_net_en, input_key_en, encoder_output)\n",
+    "                \n",
+    "                # Decoder inference per set of frames\n",
+    "                # Wait for sample duration to work with decoder model\n",
+    "                if len(encoder_output) == sample_duration:\n",
+    "                    decoded_labels, decoded_top_probs = decoder(encoder_output, exec_net_de, input_key_de)\n",
+    "                                        \n",
+    "                # Inference has finished ... Let's to display results\n",
+    "                stop_time = time.time()\n",
+    "                \n",
+    "                # Calculate processing time\n",
+    "                processing_times.append(stop_time - start_time)\n",
+    "                # Create a text template to show inference results over video\n",
+    "                text_template = '{label},{conf:.2f}%,{Time:.1f}ms,{fps:.1f}FPS'\n",
+    "                # Use processing times from last 200 frames\n",
+    "                if len(processing_times) > 200:\n",
+    "                    processing_times.popleft()\n",
+    "                    \n",
+    "                _, f_width = frame.shape[:2]\n",
+    "                # Mean processing time [ms]\n",
+    "                processing_time = np.mean(processing_times) * 1000\n",
+    "                fps = 1000 / processing_time\n",
+    "\n",
+    "                # Visualize the results\n",
+    "                for i in range(0, 3):\n",
```

Comment:
Put inference time and FPS in the first line and then results with probability e.g.

Inference time: 20.0ms (50 FPS)
drinking beer, 80%
drinking juice, 18%
sleeping, 1%